### PR TITLE
Feature/bloc andcubit

### PR DESCRIPTION
**Pull request by SolMendiola**
## :link: Notion:

- [Exercise - TODO App](https://www.notion.so/xmartlabs/Exercise-TODO-App-7c22250bc96d440a84814e14abefabe6)

---

## :pencil2: Description:

Change stateful widgets to cubit and add functionalities to the bottoms.
---

### CODE REVIEW

File: to_do_app/lib/detailTask/detail_task.dart [hunk 1]
```diff
@@ -1,49 +1,89 @@
...
+            Icon(Icons.arrow_back_ios, color: Colors.white),
+            Text('Todos', style: TextStyle(fontSize: 17, color: Colors.white)),
+          ]),
+        ),
+      ),
       body: Container(
-        width: 375,
-        height: 262,
+        width: double.infinity,
+//        height: 262,
```
Comment (f7deleon) [inline to_do_app/lib/detailTask/detail_task.dart]: remove this

File: to_do_app/lib/detailTask/detail_task_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    });
+  }
+
+  CardToDo getTask() {
+    return repo.getTask(index);
+  }
+
+  void doneButtom() {
+    repo.update(index, !state.task.done);
+  }
```
Comment (f7deleon) [inline to_do_app/lib/detailTask/detail_task_cubit.dart]: return with `=>`

File: to_do_app/lib/home/home_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    taskRepository.getTasksStream().listen((event) => emit(state.copyWith(tasks: event)));
+  }
+
+  void onChanged(int index, bool done) {
+    taskRepository.update(index, done);
+  }
+
+  void clearAllDone(){
+    taskRepository.clearAllDone();
+  }
```
Comment (f7deleon) [inline to_do_app/lib/home/home_cubit.dart]: return with `=>`

File: to_do_app/lib/detailTask/detail_task_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  final repo = RepositoryTask();
+  final CardToDo element;
+  final int index;
+  StreamSubscription<CardToDo>? subscription;
+
+  DetailTaskCubit(this.index)
+      : element = RepositoryTask().getTask(index),
+        super(TaskState(task: RepositoryTask().getTask(index))) {
+    subscription = repo.getTaskStream(index).listen((event) {
+      print(event.toString());
```
Comment (f7deleon) [inline to_do_app/lib/detailTask/detail_task_cubit.dart]: we usually don't keep prints on the final code unless we want to record a failure or a specific behavior, remove it

File: to_do_app/lib/addTask/add_task_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    ));
+  }
+
+  void changeTitle(String title){
+    emit(state.copyWith(
+      task: state.task.copyWith(
+        title: title,
+      )
+    ));
+  }
```
Comment (f7deleon) [inline to_do_app/lib/addTask/add_task_cubit.dart]: maybe use `=>` return?

**Review by f7deleon — APPROVED** [review]
Nice!

**Review by sante — APPROVED** [review]

**Pull request merged**